File: Python/Program/PetShop.py
class PetShop:
    class Produk:
        def __init__(self, id, nama, kategori, harga):
            self.__id = id
            self.__nama = nama
            self.__kategori = kategori
            self.__harga = harga

        # Getter
        def get_id(self):
            return self.__id

        def get_nama(self):
            return self.__nama

        def get_kategori(self):
            return self.__kategori

        def get_harga(self):
            return self.__harga
        
        # Setter

        def set_nama(self, nama):
            self.__nama = nama

        def set_kategori(self, kategori):
            self.__kategori = kategori

        def set_harga(self, harga):
            self.__harga = harga

        def display_info(self):
            return f"ID: {self.__id}, Nama: {self.__nama}, Kategori: {self.__kategori}, Harga: {self.__harga}"

    def __init__(self):
        self.__daftar_produk = []

    def tambah_produk(self, id, nama, kategori, harga):
        produk = self.Produk(id, nama, kategori, harga)
        self.__daftar_produk.append(produk)
        print("Produk PetShop berhasil ditambahkan!")

    def tampilkan_produk(self):
        print("\nDaftar Produk PetShop:")
        for produk in self.__daftar_produk:
            print(produk.display_info())

    def hapus_produk(self, id):
        for produk in self.__daftar_produk:
            if produk.get_id() == id:
                self.__daftar_produk.remove(produk)
                print("Produk berhasil dihapus!")
                return
        print(f"Produk dengan ID {id} tidak ditemukan.")

    def ubah_produk(self, id, nama, kategori, harga):
        for produk in self.__daftar_produk:
            if produk.get_id() == id:
                produk.set_nama(nama)
                produk.set_kategori(kategori)
                produk.set_harga(harga)
                print("Produk berhasil diubah!")
                return
        print(f"Produk dengan ID {id} tidak ditemukan.")

    def cari_produk(self, nama):
        for produk in self.__daftar_produk:
            if produk.get_nama().lower() == nama.lower():
                print(produk.display_info())
                return
        print(f"Produk dengan nama {nama} tidak ditemukan.")

File: PHP/Program/Petshop.php
<?php
session_start();

class Produk {
    private $id;
    private $nama;
    private $kategori;
    private $harga;
    private $foto;

    // Constructor
    public function __construct($id, $nama, $kategori, $harga, $foto) {
        $this->id       = $id;
        $this->nama     = $nama;
        $this->kategori = $kategori;
        $this->harga    = $harga;
        $this->foto     = $foto;
    }

    // Getter
    public function getId() {
        return $this->id;
    }
    public function getNama() {
        return $this->nama;
    }
    public function getKategori() {
        return $this->kategori;
    }
    public function getHarga() {
        return $this->harga;
    }
    public function getFoto() {
        return $this->foto;
    }

    // Setter
    public function setNama($nama) {
        $this->nama = $nama;
    }
    public function setKategori($kategori) {
        $this->kategori = $kategori;
    }
    public function setHarga($harga) {
        $this->harga = $harga;
    }
    public function setFoto($foto) {
        $this->foto = $foto;
    }
}

class PetShop {
    public function __construct() {
        if (!isset($_SESSION['produk'])) {
            $_SESSION['produk'] = [];
        }
    }

    public function tambahProduk($id, $nama, $kategori, $harga, $foto) {
        $produkBaru = new Produk($id, $nama, $kategori, $harga, $foto);
        $_SESSION['produk'][] = $produkBaru;
        $_SESSION['pesan'] = "Produk berhasil ditambahkan!";
    }

    public function tampilkanProduk() {
        echo "<h2 style='color: blue;'>Daftar Produk PetShop</h2>";
        if (empty($_SESSION['produk'])) {
            echo "<p style='color: red;'>Tidak ada produk yang tersedia.</p>";
            return;
        }

        echo "<table border='1' style='width: 100%; text-align: left; background-color: #f9f9f9;'>";
        echo "<tr style='background-color: #add8e6;'>
                <th>ID</th>
                <th>Nama</th>
                <th>Kategori</th>
                <th>Harga</th>
                <th>Foto</th>
              </tr>";
        foreach ($_SESSION['produk'] as $p) {
            if ($p instanceof Produk) {
                echo "<tr>
                        <td>" . $p->getId() . "</td>
                        <td>" . $p->getNama() . "</td>
                        <td>" . $p->getKategori() . "</td>
                        <td>" . $p->getHarga() . "</td>
                        <td><img src='uploads/" . $p->getFoto() . "' width='100'></td>
                      </tr>";
            }
        }
        echo "</table>";
    }

    public function hapusProduk($id) {
        foreach ($_SESSION['produk'] as $key => $p) {
            if ($p instanceof Produk && $p->getId() == $id) {
                unset($_SESSION['produk'][$key]);
                $_SESSION['pesan'] = "Produk berhasil dihapus!";
                return;
            }
        }
        $_SESSION['pesan'] = "Produk tidak ditemukan!";
    }

    public function ubahProduk($id, $nama, $kategori, $harga, $foto) {
        foreach ($_SESSION['produk'] as $p) {
            if ($p instanceof Produk && $p->getId() == $id) {
                $p->setNama($nama);
                $p->setKategori($kategori);
                $p->setHarga($harga);
                $p->setFoto($foto);
                $_SESSION['pesan'] = "Produk berhasil diubah!";
                return;
            }
        }
        $_SESSION['pesan'] = "Produk tidak ditemukan!";
    }

    public function cariProduk($nama) {
        $hasil = [];
        foreach ($_SESSION['produk'] as $p) {
            if ($p instanceof Produk && stripos($p->getNama(), $nama) !== false) {
                $hasil[] = $p;
            }
        }
        
        if (!empty($hasil)) {
            echo "<h2 style='color: green;'>Hasil Pencarian Produk</h2>";
            echo "<table border='1' style='width: 100%; text-align: left; background-color: #f9f9f9;'>";
            echo "<tr style='background-color: #add8e6;'>
                    <th>ID</th>
                    <th>Nama</th>
                    <th>Kategori</th>
                    <th>Harga</th>
                    <th>Foto</th>
                  </tr>";
            foreach ($hasil as $p) {
                echo "<tr>
                        <td>" . $p->getId() . "</td>
                        <td>" . $p->getNama() . "</td>
                        <td>" . $p->getKategori() . "</td>
                        <td>" . $p->getHarga() . "</td>
                        <td><img src='uploads/" . $p->getFoto() . "' width='100'></td>
                      </tr>";
            }
            echo "</table>";
        } else {
            echo "<p style='color: red;'>Produk tidak ditemukan!</p>";
        }
    }
}
?>
